src/Button/index.tsx

```diff
@@ -1,32 +1,57 @@
 import React from 'react';
-import PropTypes from 'prop-types';
+import PropTypes, { type Requireable } from 'prop-types';
 import classNames from 'classnames';
-import BaseButton from 'react-bootstrap/Button';
-import BaseButtonGroup from 'react-bootstrap/ButtonGroup';
-import BaseButtonToolbar from 'react-bootstrap/ButtonToolbar';
+import BaseButton, { type ButtonProps as BaseButtonProps } from 'react-bootstrap/Button';
+import BaseButtonGroup, { type ButtonGroupProps as BaseButtonGroupProps } from 'react-bootstrap/ButtonGroup';
+import BaseButtonToolbar, { type ButtonToolbarProps } from 'react-bootstrap/ButtonToolbar';
+import { type BsPrefixRefForwardingComponent as ComponentWithAsProp } from 'react-bootstrap/esm/helpers';
```

Comment:
[question] Given this pattern of importing `BsPrefixRefForwardingComponent` (and `BsPrefixProps`, see https://github.com/openedx/paragon/pull/3082/), I wonder if there's a more common/global location to define these types when renaming them so each individual component doesn't need to do the same re-naming repeatedly.

For example, `Hyperlink` (in https://github.com/openedx/paragon/pull/3082/) also relies on this pattern so it'd be nice to only rename it in one place.

Reply:
Yes, that's a good idea and I'm happy to move it somewhere shared. Do you have a place in mind or want me to make something up?

Reply:
First thing that came to my mind was to rename https://github.com/openedx/paragon/tree/master/src/utils/propTypes to `src/utils/types` and then put it in there, but I'm very open to any other ideas.